compiler/src/codegen/transl_anf.re

```diff
@@ -1310,63 +1310,82 @@ let transl_signature = (~functions, ~imports, signature) => {
       },
     imports.specs,
   );
-  let sign =
-    List.map(
-      fun
-      | TSigValue(vid, {val_repr, val_internalpath} as vd) => {
-          let id =
-            switch (val_internalpath) {
-            | PIdent(id) => id
-            | PExternal(_) =>
-              switch (Path_tbl.find_opt(imports.path_map, val_internalpath)) {
-              | Some(id) => id
-              | None =>
-                failwith(
-                  "Impossible: path to import not found "
-                  ++ Path.name(val_internalpath),
-                )
-              }
-            };
+  let rec process_item = (~prefix="", item) => {
+    switch (item) {
+    | TSigValue(vid, {val_repr, val_internalpath} as vd) =>
+      let id =
+        switch (val_internalpath) {
+        | PIdent(id) => id
+        | PExternal(_) =>
+          switch (Path_tbl.find_opt(imports.path_map, val_internalpath)) {
+          | Some(id) => id
+          | None =>
+            failwith(
+              "Impossible: path to import not found "
+              ++ Path.name(val_internalpath),
+            )
+          }
+        };
+      exports :=
+        [
+          GlobalExport({
+            ex_global_name: prefix ++ Ident.name(vid),
+            ex_global_internal_name: Ident.unique_name(id),
+          }),
+          ...exports^,
+        ];
+      switch (val_repr) {
+      | ReprFunction(args, rets, _) =>
+        switch (Ident_tbl.find_opt(func_map, id)) {
+        | Some(internal_name) =>
+          let external_name = Ident.name(vid);
           exports :=
             [
-              GlobalExport({
-                ex_global_name: Ident.name(vid),
-                ex_global_internal_name: Ident.unique_name(id),
+              FunctionExport({
```

Comment:
Same as `GlobalExport`.

Reply:
Why do we call these `FunctionExport` instead of `WasmExport` or even `WasmFunctionExport`? Is there a difference between a wasm function export and wasm global export?